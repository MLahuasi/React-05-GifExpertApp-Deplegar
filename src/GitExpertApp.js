import React, { useState } from 'react';
import { AddCategoria } from './components/AddCategoria';
import { GifGrid } from './components/GifGrid';

const GitExpertApp = ( { defaultCategorias = [] } ) => {
     //Cargar categor√≠as para que sean renderizadas por React
    //const [categorias, setCategorias] = useState(['Dragon Ball']);
    const [categorias, setCategorias] = useState(defaultCategorias);
    return (
        <>
            <h1>GitExpertApp</h1>
            <AddCategoria setCategorias={setCategorias} />
            <hr />
            <ol>
                {
                    /**
                     * Llamar a Hook GifGrid
                     */
                    categorias.map(categoria => (
                        <GifGrid
                            key={categoria}
                            categoria={categoria}
                        />))
                }
            </ol>

        </>
    );
}

export default GitExpertApp;


