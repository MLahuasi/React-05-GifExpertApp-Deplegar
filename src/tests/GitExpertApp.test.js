import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GitExpertApp from '../GitExpertApp'

describe('Test GitExpertApp', () => {

    let wrapper = shallow(<GitExpertApp />)

    test('Debe realizar una captura ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const lista = ['metallica', 'iron maiden','megadeth'];

        let wrapper = shallow(<GitExpertApp defaultCategorias={lista}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(lista.length);
    });

})
