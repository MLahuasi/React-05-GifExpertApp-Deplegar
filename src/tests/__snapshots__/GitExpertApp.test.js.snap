// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GitExpertApp
  </h1>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="metallica"
      key="metallica"
    />
    <GifGrid
      categoria="iron maiden"
      key="iron maiden"
    />
    <GifGrid
      categoria="megadeth"
      key="megadeth"
    />
  </ol>
</Fragment>
`;

exports[`Test GitExpertApp Debe realizar una captura  1`] = `
<Fragment>
  <h1>
    GitExpertApp
  </h1>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
